routing/router.go

```diff
@@ -1309,8 +1326,17 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 		channelID := lnwire.NewShortChanIDFromInt(msg.ChannelID)
 		fundingTx, err := r.fetchFundingTx(&channelID)
 		if err != nil {
-			return errors.Errorf("unable to fetch funding tx for "+
-				"chan_id=%v: %v", msg.ChannelID, err)
+			// If the funding transaction isn't found at all, then
+			// we'll mark the edge itself as a zombie so we don't
+			// continue to request it. We use the "zero key" for
+			// both node pubkeys so this edge can't be resurrected.
+			zErr := r.addZombieEdge(msg.ChannelID)
```

Comment:
Similar to the error handling for `GetUtxo`, what about `GetBlockHash` and `GetBlock`? Unlikely to fail in most cases, but there's a slight gap where if the backend becomes unavailable, the health check hasn't fired yet, and a wave of new gossip arrives, then we'll never be able to process/resurrect those channels. We don't have a consistent error for `block not found` across all backends, so might be a bit more involved.

Reply:
I think we can account for this by specifying the pubkeys as normal, this way we'll eventually hear about the channels again causing them to be resurrected.

Reply:
Pushed up a new commit to pattern match on errors based on commonalities between `btcd` the `bitcoind` errors.